src/session.ts

```diff
@@ -665,6 +711,32 @@ export class Session extends EventEmitter {
 
     let url = this.url
 
+    if (this.customUrlHandler && !this.url) {
+      const customUrlHandlerPath = path.resolve(
+        process.cwd(),
+        this.customUrlHandler,
+      )
+      if (!fs.existsSync(customUrlHandlerPath)) {
+        log.error(`Custom error handler not found: "${customUrlHandlerPath}"`)
+        throw new Error(
+          `Custom error handler not found: "${customUrlHandlerPath}"`,
+        )
+      }
+      this.customUrlHandlerFn = (
```

Comment:
Can we move this assignment into the class constructor?

Reply:
We can do it, but it would be slightly complicated because importing is an asynchronous process, so it will increase code complications in the session class. 
If you find this change important, I can figure something out, but I don't understand the necessity of this.